src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,349 @@
+ */
+public class Duke extends Application {
+
+    /**
+     * Represents the type of command
+     */
+    public enum CommandType {
+        LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND, BYE
+    }
+
+    /**
+     * Represents the storage of tasks
+     */
+    private Storage storage;
+
+    /**
+     * Represents the list of tasks
+     */
+    private TaskList tasks;
+
+    /**
+     * Represents the user interface
+     */
+    private Ui ui;
+
+    /**
+     * Represents the parser
+     */
+    private Parser parser;
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/Dauser.jpg"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/Daduke.jpg"));
+
+
+    /**
+     * Constructor for Duke
+     */
+    public Duke() {
+        ui = new Ui("KokBot");
+        storage = new Storage(Paths.get("data", "duke.txt"));
+        parser = new Parser();
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Constructor for Duke
+     *
+     * @param filePath Path of the file
+     * @param botName  Name of the bot
+     */
+    public Duke(Path filePath, String botName) {
+        ui = new Ui(botName);
+        storage = new Storage(filePath);
+        parser = new Parser();
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    @Override
+    public void start(Stage stage) {
+
+        // Step 1. Setting up required components
+
+        // The container for the content of the chat to scroll.
+        scrollPane = new ScrollPane();
+        dialogContainer = new VBox();
+        scrollPane.setContent(dialogContainer);
+
+        userInput = new TextField();
+        sendButton = new Button("Send");
+
+        AnchorPane mainLayout = new AnchorPane();
+        mainLayout.getChildren().addAll(scrollPane, userInput, sendButton);
+
+        scene = new Scene(mainLayout);
+
+        stage.setScene(scene);
+        stage.show();
+
+        // Step 2. Formatting the window to look as expected
+        stage.setTitle("Duke");
+        stage.setResizable(false);
+        stage.setMinHeight(600.0);
+        stage.setMinWidth(400.0);
+
+        mainLayout.setPrefSize(400.0, 600.0);
+
+        scrollPane.setPrefSize(385, 535);
+        scrollPane.setHbarPolicy(ScrollPane.ScrollBarPolicy.NEVER);
+        scrollPane.setVbarPolicy(ScrollPane.ScrollBarPolicy.ALWAYS);
+
+        scrollPane.setVvalue(1.0);
+        scrollPane.setFitToWidth(true);
+
+        // You will need to import `javafx.scene.layout.Region` for this.
```

Comment:
Perhaps remove these comments since the relevant components are already imported above.